scalar_udf.go

```diff
@@ -0,0 +1,241 @@
+package duckdb
+
+// Related issues: https://golang.org/issue/19835, https://golang.org/issue/19837.
+
+/*
+#include <duckdb.h>
+
+void scalar_udf_callback(duckdb_function_info, duckdb_data_chunk, duckdb_vector);
+void scalar_udf_delete_callback(void *);
+
+typedef void (*scalar_udf_callback_t)(duckdb_function_info, duckdb_data_chunk, duckdb_vector);
+*/
+import "C"
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"runtime/cgo"
+	"unsafe"
+)
+
+type ScalarFuncConfig interface {
+	InputTypeInfos() []TypeInfo
+	ResultTypeInfo() TypeInfo
+}
```

Comment:
Would this not be better as a struct? If in the future we would need to add some method to this it would be a breaking change, adding a struct field wouldn't have to be as breaking.

Reply:
That's a good point. 👍